rules-reviewed/openshift/local-storage.windup.xml

```diff
@@ -83,17 +89,52 @@
                 <filecontent pattern="{path2}" filename="{*}"/>
             </when>
             <perform>
-                <hint title="Detected URL with local file system path" effort="1" category-id="potential">
-                    <message>Detected URL with local file system path.
-                    
-                    Local files in containers are not permanent storage and can disappear when an instance of the application terminates. Avoid accessing local filesystems from applications running in OpenShift environments, and consider using a database or cloud-based storage system instead.
+                <hint title="Detected URL with local file system path" effort="1" category-id="openshift-potential">
+                    <message>
+                    Detected URL with local file system path.  
+                    Local files in containers are not permanent storage and can disappear when an instance of the application terminates.  
+                    Avoid accessing local filesystems from applications running in OpenShift environments, and consider using a database or cloud-based storage system instead.
                     </message>
+                    <link href="https://docs.openshift.com/container-platform/3.4/architecture/additional_concepts/storage.html" title="OpenShift with persistent storage (Concepts)"></link>
                 </hint>
             </perform>
             <where param="path2">
                 <!-- URL with local path -->
                 <matches pattern="file://" />
             </where>
         </rule>
+        <rule id="local-storage-00005">
+            <when>
+              <or>
+                  <javaclass references="java.nio.channels.AsynchronousFileChannel{*}" />
+                  <javaclass references="java.nio.channels.FileChannel{*}" />
+                  <javaclass references="java.nio.channels.FileLock{*}" />
+                  <javaclass references="java.nio.file.DirectoryStream{*}" />
+                  <javaclass references="java.nio.file.Files{*}" />
+                  <javaclass references="java.nio.file.FileStore{*}" />
+                  <javaclass references="java.nio.file.FileSystem{*}" />
+                  <javaclass references="java.nio.file.FileSystems{*}" />
+                  <javaclass references="java.nio.file.Path{*}" />
+                  <javaclass references="java.nio.file.Paths{*}" />
+              </or>
+            </when>
+            <perform>
+              <iteration>
+                <hint title="Java APIs for local file system (java.nio classes)" effort="3" category-id="openshift-optional">
```

Comment:
We have the same recommendation and hint.

It does not make a difference for the recommended action if the NIO or Java IO APIs are used.